state/machine.go

```diff
@@ -2203,6 +2182,78 @@ func (m *Machine) RemoveUpgradeSeriesLock() error {
 	return nil
 }
 
+// UpgradeSeriesStatus returns the status of a series upgrade.
+func (m *Machine) UpgradeSeriesStatus(unitName string) (params.UnitSeriesUpgradeStatus, error) {
+	coll, closer := m.st.db().GetCollection(machineUpgradeSeriesLocksC)
+	defer closer()
+
+	var lock upgradeSeriesLock
+	err := coll.FindId(m.Id()).One(&lock)
+	if err == mgo.ErrNotFound {
+		return "", errors.NotFoundf("upgrade series lock for machine %q", m.Id())
+	}
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	for _, unit := range lock.PrepareUnits {
+		if unit.Id == unitName {
+			return unit.Status, nil
+		}
+	}
+
+	return "", errors.NotFoundf("unit %q of machine %q", unitName, m.Id())
+}
+
+// SetUpgradeSeriesStatus sets the status of a series upgrade.
+func (m *Machine) SetUpgradeSeriesStatus(unitName string, status params.UnitSeriesUpgradeStatus) (string, error) {
+	buildTxn := func(attempt int) ([]txn.Op, error) {
+		if attempt > 0 {
+			if err := m.Refresh(); err != nil {
+				return nil, errors.Trace(err)
+			}
+		}
+		locked, err := m.IsLocked()
```

Comment:
Calling IsLocked() is here appears to be redundant given the code at  2227-2233 with an extra error.